daemon/health.go

```diff
@@ -188,15 +188,27 @@ func handleProbeResult(d *Daemon, c *container.Container, result *types.Healthch
 func monitor(d *Daemon, c *container.Container, stop chan struct{}, probe probe) {
 	probeTimeout := timeoutWithDefault(c.Config.Healthcheck.Timeout, defaultProbeTimeout)
 	probeInterval := timeoutWithDefault(c.Config.Healthcheck.Interval, defaultProbeInterval)
+loop:
 	for {
 		select {
 		case <-stop:
 			logrus.Debugf("Stop healthcheck monitoring for container %s (received while idle)", c.ID)
 			return
 		case <-time.After(probeInterval):
-			logrus.Debugf("Running health check for container %s ...", c.ID)
 			startTime := time.Now()
+			healthChecksWaiters.Inc()
 			ctx, cancelProbe := context.WithTimeout(context.Background(), probeTimeout)
```

Comment:
Blocking on the limiter shouldn't have an effect on probe timeout.

Reply:
Why not? This cost would be incurred if it were dogpiling syscalls or waiting on a semaphore.

Reply:
Probe timeout is what `Dockerfile` author or user starting a container sets depending on the specifics of the application and the health check command. Eg. a reliable healthcheck shouldn't have a long timeout. These users can't predict any scheduling constraints or how the configuration of other containers will start to have an effect on this value.